runtime/compiler/control/J9Options.cpp

```diff
@@ -2314,6 +2314,12 @@ bool J9::Options::feLatePostProcess(void * base, TR::OptionSet * optionSet)
       self()->setOption(TR_DisableNextGenHCR);
       }
 
+#if !defined(TR_HOST_X86)
+   //The bit is set when -XX:+JITInlineWatches is specified
+   if (J9_ARE_ANY_BITS_SET(javaVM->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_JIT_INLINE_WATCHES))
+      TR_ASSERT_FATAL(false, "this platform doesn't support JIT inline field watch");
```

Comment:
I'm not sure a fatal assert and crash is the right thing to do - isn't there a more graceful way to fail?

Reply:
It's a new option for users to experiment with and will not be publicized so It's probably better to keep the assertion here and let it fail fast in our testing environment.